src/linked_list.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding utf-8 -*-
+
+
+class Node(object):
+
+    def __init__(self, data=None, next_node=None):
+        self.data = data
+        self.next_node = next_node
+
+    def get_data(self):
+        return self.data
+
+    def get_next(self):
+        return self.next_node
+
+    def set_next(self, next_node):
+        self.next_node = next_node
+
+
+class LinkedList(object):
+
+    def __init__(self, head=None):
+        self.head = head
+
+    def push(self, val):
+        new_node = Node(val)
+        new_node.set_next(self.head)
+        self.head = new_node
+
+    def pop(self):
+        try:
+            returned_value = self.head.get_data()
+        except AttributeError:
+            raise IndexError('can not pop from empty LinkedList')
+        self.head = self.head.get_next()
+        return returned_value
+
+    def size(self):
+        current = self.head
+        count = 0
+        while current:
+            count += 1
+            current = current.get_next()
+        return count
+
+    def search(self, val):
+        current = self.head
+        found = False
+        while current and found is False:
+            if current.get_data() == val:
+                found = True
+            else:
+                current = current.get_next()
+        if current is None:
+            # raise ValueError("that value is not in the list")
+            return None
```

Comment:
If `current` is in fact None, then you could just `return current` like on line 57 anyway. You probably only need one `return` statement in this function.